ably/hosts_test.go

```diff
@@ -0,0 +1,284 @@
+package ably_test
+
+import (
+	"testing"
+
+	"github.com/ably/ably-go/ably"
+	"github.com/ably/ably-go/ably/internal/ablyutil"
+)
+
+func Test_RSC15_RestHostFallback(t *testing.T) {
+	t.Parallel()
+
+	t.Run("RSC15a: should get fallback hosts in random order", func(t *testing.T) {
+		clientOptions := ably.NewClientOptions()
+		restHosts := ably.NewRestHosts(clientOptions)
+		// All expected hosts supposed to be tried upon
+		expectedHosts := []string{
+			"rest.ably.io",
+			"a.ably-realtime.com",
+			"b.ably-realtime.com",
+			"c.ably-realtime.com",
+			"d.ably-realtime.com",
+			"e.ably-realtime.com",
+		}
+
+		// Get first preferred restHost
+		var actualHosts []string
+		prefHost := restHosts.GetPreferredHost()
+		assertEquals(t, "rest.ably.io", prefHost)
+
+		actualHosts = append(actualHosts, prefHost)
+
+		// Get all fallback hosts in random order
+		actualHosts = append(actualHosts, restHosts.GetAllRemainingFallbackHosts()...)
+
+		assertNotDeepEquals(t, expectedHosts, actualHosts)
+		assertDeepEquals(t, ablyutil.Sort(expectedHosts), ablyutil.Sort(actualHosts))
```

Comment:
Given this test is called "should get fallback hosts in random order", should it be asserting that the hosts are actually random and not in alphabetical order?

Reply:
`assertNotDeepEquals` makes sure they are random

Reply:
I think a better approach is to use an explicit seed for the random generator that leads to a well known order, and then assert that order here, since not controlling randomness in tests typically leads to flaky tests.

Reply:
I will check, currently I never faced failing tests related to shuffled hosts

Reply:
I have removed unnecessary assertions to check shuffle, since we already have assertions in place for shuffle method. Also, I do not want deterministic order for shuffle, since it destroys purpose of having shuffle. I do not see any flakiness nature of the tests as of now, for ~40-50 runs.  I will change the tests, if I found them flaky in the future (highly unlikely).